lips/lip-10.md

```diff
@@ -0,0 +1,191 @@
+  * Libra ID is a user-friendly identifier, unlike lip-5 address
+  * Libra ID resolution process can allow to exchange information about both sender and receiver for payment
+  * Libra ID resolution allows to fetch details about each participant, like profile picture
+  * Libra ID allows for better anonymity on-chain, as it is possible to share it once, but for wallet apps to transparently resolve it to new subaddress for each transaction. This is a big improvement comparing to exchange of lip-5 identifiers, that leaks privacy information on the chain and allows to track all transactions for the individual subaccount.
+* For the merchant use case, Libra ID allows users to self-identify themself to merchants. In this case Libra ID plays the role of credit card number or also can be seen as a variant of OpenID-type protocol.
+
+## Examples in other blockchains
+Ripple proposes PayID. PayID has some similarity to the current proposal, but is not entirely the same thing.
+For example, PayID seems to target cross-chain payments, while libra id is supposed to facilitate cross-vasp payments, but only inside the libra network.
+
+# Libra ID format
+Libra ID is a string in format `<user-id>$<libra-id-domain>`.
+
+* **user-id**, identifies an account within a VASP:
+  * This part is case insensitive.
+  * Regex: `^[a-zA-Z0-9][a-zA-Z0-9-_+.]?$`
+  * This part is not published on the chain, there is no way for an external entity to list all the users within VASP.
+* **libra-id-domain** identifies VASP:
+  * This part is case insensitive.
+  * Regex: `^[a-zA-Z0-9][a-zA-Z0-9-]?$`
+  * libra-id-domains are published on the chain
+  * Each libra-id-domain is uniquely associated with single VASP
+  * VASP can have multiple libra-id-domains associated, thought it is not expected to be common
+
+## Multiple accounts per user
+VASP should not assume that Libra ID uniquely identifies the user within the system. User that have accounts with different VASP would have different Libra ID within each VASP.
+
+Furthermore, a user can have multiple Libra IDs within the same VASP. This does not affect the protocol, but VASP can leverage this, for example to allow users to create multiple ‘accounts’ like checking, savings, etc.
+
+VASP can also allow users to generate temporary / situational Libra IDs. This can be used in many scenarios such as:
+
+* Temporary Libra ID for some merchant payments, when user don’t want to share their ‘main’ Libra ID with the merchant
+* Publicly shared Libra IDs for the various campaigns, like donations, etc.
+
+## Libra ID domains
+Each libra ID domain is associated with a VASP. The source of truth for this association is a blockchain - a special LibraIdDomains resource must be published into VASP for it to expose it’s LibraID domain.
+
+```
+resource struct LibraIdDomains {
+    domains: vector<LibraIdDomain>,
+}
+
+struct LibraIdDomain {
+    domain: vector<u8>,
+    wallet_uri: vector<u8>,
```

Comment:
Why vec<u8> instead of strings? Or at least what encoding are the bytes?  ASCII?

Reply:
Afaik there is simply no string type in move

Reply:
Added clarification on the encoding